nos/server/http/_service.py

```diff
@@ -119,40 +120,111 @@ def get_client() -> Client:
         """Check if the server is alive."""
         return JSONResponse(
             content={"status": "ok" if client.IsHealthy() else "not_ok"}, status_code=status.HTTP_200_OK
         )
 
-    @app.post("/infer", status_code=status.HTTP_201_CREATED)
+    @app.post(f"/{version}/infer", status_code=status.HTTP_201_CREATED)
     def infer(
         request: InferenceRequest,
         client: Client = Depends(get_client),
     ) -> JSONResponse:
+        """Perform inference on the given input data.
+
+        $ curl -X "POST" \
+            "http://localhost:8000/v1/infer" \
+            -H "Content-Type: appication/json" \
+            -d '{
+                "model_id": "yolox/small",
+                "inputs": {
+                    "images": ["data:image/jpeg;base64,..."],
+                }
+            }'
+
+        $ curl -X "POST" \
+            "http://localhost:8000/v1/infer" \
+            -H "Content-Type: application/json" \
+            -d '{
+                "model_id": "openai/clip",
+                "method": "encode_text",
+                "inputs": {
+                    "texts": ["fox jumped over the moon"]
+                }
+            }'
+
+        """
         logger.debug(f"Decoding input dictionary [model={request.model_id}, method={request.method}]")
         request.inputs = decode_item(request.inputs)
         return _infer(request, client)
 
-    @app.post("/infer_file", status_code=status.HTTP_201_CREATED)
+    @app.post(f"/{version}/infer/file", status_code=status.HTTP_201_CREATED)
     def infer_file(
         model_id: str = Form(...),
-        method: str = Form(None),
-        file: UploadFile = File(...),
+        method: Optional[str] = Form(None),
+        file: Optional[UploadFile] = File(None),
+        url: Optional[str] = Form(None),
         client: Client = Depends(get_client),
     ) -> JSONResponse:
-        """Perform inference on the given input data."""
+        """Perform inference on the given input data using multipart/form-data.
+
+        $ curl -X POST \
+            'http://localhost:8000/v1/infer/file \
+            -H 'accept: application/json' \
+            -H 'Content-Type: multipart/form-data' \
+            -F 'model_id=yolox/small' \
+            -F 'file=@test.jpg'
+
+        $ curl -X POST \
+            'http://localhost:8000/v1/infer/file \
+            -H 'accept: application/json' \
+            -H 'Content-Type: multipart/form-data' \
+            -F 'model_id=yolox/small' \
+            -F 'url=https://.../test.jpg'
+
+        """
+        logger.debug(f"Received inference request [model={model_id}, method={method}]")
+        request = InferenceRequest(model_id=model_id, method=method, inputs={})
+        file_object: SpooledTemporaryFile = None
+        file_basename: str = None
+        if file is not None:
+            assert isinstance(file.file, SpooledTemporaryFile), f"Invalid file [file={file}]"
```

Comment:
[nit] might eventually want to check the filesize here and set a reasonable maximum

Reply:
Agreed. Will probably add a bunch of these checks on the HTTP gateway side.